packages/sfui/frameworks/react/vite.config.ts

```diff
@@ -35,6 +35,7 @@ export default defineConfig({
       output: {
         preserveModules: true,
         preserveModulesRoot: './',
+        banner: `'use client';`,
```

Comment:
Maybe I didn't use a correct phrasing with "server components". Without the `use client` directive, a component will be a shared component. It may be used within client and server context. We don't have in SFUI any server components which should be rendered _only_ on the server side.

For example when we have a SfBadge, then

- when it will be used only within server components, its code won't be included in the client side bundle, the whole code of SfBadge will be returned to the browser as [React Server Component Payload](https://nextjs.org/docs/app/building-your-application/rendering/server-components#how-are-server-components-rendered)
- when it will be used also in client components, its code will be included in client side bundle

https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#advice-for-library-authors

Reply:
I was about to say the same thing once I've seen this PR. 
One thing is that not all components are client side only, the other thing is that we didn't wanted to place `use client` or `use server` because its not `react` standard, only next (at least it was 1 year ago), and `next` is one of many meta-frameworks. Those components should work in `gatsby`, `remix` as well. Not sure how standard looks right now and how components with those directives works on other meta-frameworks. 
So if we would even consider placing `use client` on selected components, we need to check how they are working on other meta-frameworks.

Reply:
it's already a React standard, so this makes things easier :D https://react.dev/reference/rsc/use-client

Reply:
at least that one thing is solved :D

Reply:
@Szymon-dziewonski @jagoral updated with **explicit** directive in interactive components, based on React docs